src/shared/logs-show.c

```diff
@@ -1228,14 +1258,234 @@ static int output_cat(
+        const void *data;
+} FormatSpecifier;
+
+static int format_specifier_field(
+                        char specifier,
+                        FILE *f,
+                        sd_journal *j,
+                        OutputFlags flags,
+                        const size_t highlight[2],
+                        const void *data) {
+
+        const char *field = ASSERT_PTR(data);
+        const void *d = NULL;
+        const char *color_on = "", *color_off = "", *highlight_on = "";
+        size_t l = 0;
+        int r;
+
+        assert(specifier);
+        assert(f);
+        assert(j);
+
+        if (specifier == 'm' && FLAGS_SET(flags, OUTPUT_COLOR))
+                get_log_colors(get_priority(j, flags), &color_on, &color_off, &highlight_on);
+
+        r = get_field_data(j, field, &d, &l);
+        if (r < 0 && !IN_SET(r, -EBADMSG, -ENOENT))
+                return r;
+
+        if (specifier == 'm' && FLAGS_SET(flags, OUTPUT_COLOR)) {
+                if (highlight) {
+                        assert(highlight[0] <= highlight[1]);
+                        assert(highlight[1] <= l);
+
+                        fputs(color_on, f);
+                        fwrite((const char*) d, 1, highlight[0], f);
+                        fputs(highlight_on, f);
+                        fwrite((const char*) d + highlight[0], 1, highlight[1] - highlight[0], f);
+                        fputs(color_on, f);
+                        fwrite((const char*) d + highlight[1], 1, l - highlight[1], f);
+                        fputs(color_off, f);
+                } else {
+                        fputs(color_on, f);
+                        fwrite((const char*) d, 1, l, f);
+                        fputs(color_off, f);
+                }
+        } else
+                fwrite((const char*) d, 1, l, f);
+
+        return 0;
+}
+
+const FormatSpecifier format_specifier_table[] = {
+        { 'b', format_specifier_field, "_BOOT_ID" },
+        { 'p', format_specifier_field, "_COMM" },
+        { 'P', format_specifier_field, "_PID" },
+        { 'U', format_specifier_field, "_UID" },
+        { 'G', format_specifier_field, "_GID" },
+        { 'H', format_specifier_field, "_HOSTNAME" },
+        { 'M', format_specifier_field, "_MACHINE_ID" },
+        { 'e', format_specifier_field, "_EXE" },
+        { 'E', format_specifier_field, "_CMDLINE" },
+        { 'n', format_specifier_field, "_SYSTEMD_UNIT" },
+        { 'N', format_specifier_field, "_NAMESPACE" },
+        { 'T', format_specifier_field, "_TRANSPORT" },
+        { 'c', format_specifier_field, "_SYSTEMD_CGROUP" },
+        { 's', format_specifier_field, "_SYSTEMD_SLICE" },
+
+        { 'm', format_specifier_field, "MESSAGE"},
+        { 'i', format_specifier_field, "SYSLOG_IDENTIFIER" },
+        { 'f', format_specifier_field, "CODE_FILE" },
+        { 'F', format_specifier_field, "CODE_FUNC" },
+        { 'l', format_specifier_field, "CODE_LINE" },
+        { 'X', format_specifier_field, "ERRNO" },
```

Comment:
I'd really stick to the field names for this, and not bother with short specifiers. that adds another level of indirection for very little reason.

Reply:
but i think it would be easier for everyone to just use the long field names we also use for filtering. Sounds weird using one type of specifier for filtering and another for display

Reply:
But let's say you want to include the data in the format, do you really want users to type out SOURCE_REALTIME_MONOTONIC or stuff like that?

In general, the timestamps need special processing and can't be specified by a single field, especially if you want to mimick the date formatting that's done by default. Ideally that'd be doable with a single specifier

Reply:
So yeah, that might make sense, use % specifiers for formatting key fields in non-literal ways. But let's keep a lid on it, i.e. only where it really makes sense. i.e. displaying raw UNIX timestamp values probably makes little sense, hence having a % specifier to format it in human readable way makes sense to me. But let's not add specifiers for stuff that would be 1:1 the original field raw data...